### PR TITLE
Clippy fixes

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This is a good start on fixing things that clippy reports.

### CODE REVIEW

File: src/libpcp/variable/memory/mod.rs [hunk 1]
```diff
@@ -246,7 +243,7 @@ mod test {
         fn from_shape(&mut self, depth: usize, shape: &TreeShape) -> usize {
             use self::TreeShape::*;
             match shape {
-                &Node(ref value, ref children) if depth > 0 => {
+                Node(value, children) if depth > 0 => {
```
Comment (ptal) [inline src/libpcp/variable/memory/mod.rs:246]: I see `ref` was removed in many places, is it by reference by default now? Instead of being by copy?

**Review by ptal — COMMENTED** [review]
Thanks! A small question below though.

File: src/libpcp/variable/memory/mod.rs (hunk 1 above)
Comment (waywardmonkeys) [reply src/libpcp/variable/memory/mod.rs:246]: This changed some years ago as part of match ergonomics: https://rust-lang.github.io/rfcs/2005-match-ergonomics.html

Comment (ptal) [reply src/libpcp/variable/memory/mod.rs:246]: My Rust knowledge is completely outdated :sweat: Thanks.

**Pull request merged**